st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -469,6 +469,58 @@ def post(self, liveaction_api, requester_user, context_string=None, show_secrets
                                                context_string=context_string,
                                                show_secrets=show_secrets)
 
+    def put(self, id, liveaction_api, requester_user, show_secrets=False):
+        """
+        Updates a single execution.
+
+        Handles requests:
+            PUT /executions/<id>
+
+        """
+        if not requester_user:
+            requester_user = UserDB(cfg.CONF.system_user.user)
+
+        from_model_kwargs = {
+            'mask_secrets': self._get_mask_secrets(requester_user, show_secrets=show_secrets)
+        }
+
+        execution_api = self._get_one_by_id(id=id, requester_user=requester_user,
+                                            from_model_kwargs=from_model_kwargs,
+                                            permission_type=PermissionType.EXECUTION_STOP)
+
+        if not execution_api:
+            abort(http_client.NOT_FOUND, 'Execution with id %s not found.' % id)
+
+        liveaction_id = execution_api.liveaction['id']
+        if not liveaction_id:
+            abort(http_client.INTERNAL_SERVER_ERROR,
+                  'Execution object missing link to liveaction %s.' % liveaction_id)
+
+        try:
+            liveaction_db = LiveAction.get_by_id(liveaction_id)
+        except:
+            abort(http_client.INTERNAL_SERVER_ERROR,
+                  'Execution object missing link to liveaction %s.' % liveaction_id)
+
+        if liveaction_db.status in LIVEACTION_COMPLETED_STATES:
+            abort(http_client.BAD_REQUEST, 'Execution is already in completed state.')
+
+        try:
+            liveaction_db = action_service.update_status(
+                liveaction_db,
+                liveaction_api.status,
```

Comment:
Ignore this comment above.

I finally got to the end of diff and saw you limited possible attributes via openapi schema - https://github.com/StackStorm/st2/pull/3461/files#diff-c6bc5bca320e32b130cd88ba7a961ad9R3956.